nuxeo-platform-spreadsheet-web/src/main/js/app/ui/spreadsheet.js

```diff
@@ -57,29 +57,95 @@ class Spreadsheet {
     this.query.depth = 'max';
     // translate directory labels
     this.query.translate('directoryEntry', 'label');
+    this.query.pageProvider = pageProviderName;
 
-    this.query.pageProvider = pageProvider;
-
-    new Layout(connection, layout, language).fetch().then((layout) => {
-      // Check which columns to display
-      var cols = (!columns) ?
-          layout.columns.filter((c) =>  c.selectedByDefault !== false)
-          :
-          columns.map((name) => layout.columns.filter((c) => c.name === name)[0]);
-      this.columns = cols
+    // set columns based on result layout
+    if (resultLayoutName) {
+      new Layout(connection, resultLayoutName, language).fetch().then((layout) => {
+        // Check which columns to display
+        let cols = (!resultColumns) ? layout.columns.filter((c) => c.selectedByDefault !== false)
+          : resultColumns.map((name) => layout.columns.filter((c) => c.name === name)[0]);
+        this.columns = cols
           // Exclude columns without widgets
           .filter((c) => c.widgets)
           // Create our columns wrapper
           .map((c) => new Column(connection, c, layout.widgets[c.widgets[0].name], this.dirtyRenderer.bind(this)))
           // Only show columns with a known widget type and with a field
           .filter((c) => c.hasSupportedWidgetType && c.field);
-    });
+      });
+
+    // or based on result columns only
+    } else {
+
+      // default columns
+      if (!resultColumns) {
```

Comment:
these default columns would be handled in the app too... ideally in the future people would be able to configure these